crypto/hpke.c

```diff
@@ -0,0 +1,2730 @@
+/*
+ * Copyright 2019-2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/**
+ * @file
+ * An OpenSSL-based HPKE implementation of RFC9180
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <openssl/ssl.h>
+#include <openssl/rand.h>
+#include <openssl/kdf.h>
+#include <openssl/evp.h>
+#include <openssl/params.h>
+#include <openssl/param_build.h>
+#include <openssl/core_names.h>
+
+/*
+ * If we're building standalone (from github.com/sftcd/happykey) then
+ * include the former, if building from github.com/sftcd/openssl) then
+ * the latter
+ */
+#ifdef HAPPYKEY
+#include "hpke.h"
+#else
+#include <openssl/hpke.h>
```

Comment:
This doesn't follow our indentation style.  I expect that this #ifdef chain will simply disappear.

Reply:
Yep - I just committed a change that takes a "polyfill" approach to getting rid of this so that the code from https://github.com/sftcd/happykey that's not needed here, but is still useful for debug there  (e.g. as/when new ciphersuites are under development that aren't in RFC9180) don't cause hiccups. Hopefully that's an ok approach.